### DISCUSSION TITLE
Why is there a disagreement between `edge_index` dimension and the `x` feature matrix dimension?

### DISCUSSION THREAD

**Discussion by mcneela** [pyg-team/pytorch_geometric#3727] [Q&A] [answered]
Has anyone dealt with an example in pytorch geometric where there is a disagreement between the `edge_index` (sparse form of the adj matrix) and the number of rows in the feature matrix `x`? I am currently working with the PPI dataset that comes bundled with pytorch geometric. The first graph in that dataset has `x` with size 1767x50 but the `edge_index` only has dimension 1750  which means that when I create the adjacency matrix by calling `to_dense_adj(adj)` it ends up being size `1750x1750`. This causes issues, because I have an attention weight adjacency matrix based on the dimensionality of `x` which has size `1767x1767` and I cannot multiply them. I assume this is because 17 nodes (1767-1750) are fully disconnected and not connected to any other nodes in the graph, but why would pytorch geometric not account for this when creating the adjacency matrix and add an extra 17 rows/columns with all 0s and one 1?

**Comment by rusty1s** [comment, marked as answer]
Yes, this is correct. As such, most of our functions that just expect an `edge_index` matrix as input also support `max_num_nodes` or `num_nodes` arguments, see [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/utils.html?highlight=to_dense_adj#torch_geometric.utils.to_dense_adj):
```python
adj = to_dense_adj(data.edge_index, max_num_nodes=data.num_nodes)
```

**Reply by mcneela** [reply]
Got it, thank you!
